compendium/DeclarativeServices/test/TestConfigPolicies.cpp

```diff
@@ -142,72 +85,29 @@ TEST_F(tServiceComponent, testOptionalConfigPolicyImmediateFalse) // DS_CAI_FTC_
-  while (i < 3) {
-    // The Update properties sends an asynchronous request to DS to
-    // update the properties. Must wait until DS is finished before we can continue.
-    // Use DS runtime service to validate the component properties. May take more
-    // than one try.
-
-    props[uniqueProp[i]] = instance[i];
-    configuration->Update(props);
-    auto result = RepeatTaskUntilOrTimeout(
-      [&compDescDTO, &compConfigs, this, &testBundle, &configInstance]() {
-        compDescDTO = dsRuntimeService->GetComponentDescriptionDTO(
-          testBundle, configInstance);
-        if (compDescDTO.name != "") {
-          compConfigs =
-            this->dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
-        }
-      },
-      [&compConfigs, &instance, &uniqueProp, &i]() -> bool {
-        if (compConfigs.size() == 1) {
-          auto properties = compConfigs.at(0).properties;
-          auto id = properties.find(uniqueProp[i]);
-          if (id != properties.end()) {
-            return (id->second == instance[i]);
-          }
-        }
-        return false;
-      });
-
-    ASSERT_TRUE(result) << "Timed out waiting for Update Configuration"
-                           "to complete.";
-    EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
-    if (i == 0)
-    {
-      EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::SATISFIED)
-        << "Component instance state should be SATISFIED";
-    }
-    else
-    {
-      EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::ACTIVE)
-        << "Component instance state should be ACTIVE";
-    }
-
-    //Request a service reference to the new component instance. This will
-    //cause DS to construct the instance with the updated properties.
-    auto instanceI = GetInstance<test::CAInterface>();
-    ASSERT_TRUE(instanceI) << "GetService failed for CAInterface";
-
-    //Confirm component instance was created with the correct properties
-    instanceProps = instanceI->GetProperties();
-    auto uniquePropI = instanceProps.find(uniqueProp[i]);
-
-    EXPECT_TRUE(uniquePropI != instanceProps.end())
-      << uniqueProp[i] << " not found in constructed instance"; //not found
-    EXPECT_EQ(uniquePropI->second, instance[i]);
-
-    ++i;
-  }
-}
+  const std::string instanceId{ "instance" };
+  props["uniqueProp"] = instanceId;
+  configuration->Update(props);
+
+  EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::ACTIVE)
+    << "Component instance state should be ACTIVE";
+
+  // Request a service reference to the new component instance. This will
+  // cause DS to construct the instance with the updated properties.
+  auto instance = GetInstance<test::CAInterface>();
+  ASSERT_TRUE(instance) << "GetService failed for CAInterface";
+
+  //Confirm component instance was created with the correct properties
+  auto instanceProps = instance->GetProperties();
```

Comment:
The reason the properties are correct is because the Modified method was called by configuration->Update. It doesn't prove that construction worked.